### DISCUSSION TITLE
Has anyone gotten rook-ceph to install on a new OKD 4.13 cluster?

### DISCUSSION THREAD

**Discussion by fortinj66** [okd-project/okd#1718] [Q&A] [answered]
I am trying to install rook-ceph  1.12 on a new OKD 4.13 cluster.  I am running into the following issues which seem to be related to the new SCC restrictions...

`Error creating: pods "rook-ceph-mon-a-canary-6cd64b48d8-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, spec.volumes[2]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, spec.volumes[3]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, spec.volumes[4]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, spec.volumes[5]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, provider restricted-v2: .containers[0].privileged: Invalid value: true: Privileged containers are not allowed, provider restricted-v2: .containers[0].hostNetwork: Invalid value: true: Host network is not allowed to be used, provider restricted-v2: .containers[0].containers[0].hostPort: Invalid value: 3300: Host ports are not allowed to be used, provider restricted-v2: .containers[0].containers[0].hostPort: Invalid value: 6789: Host ports are not allowed to be used, provider restricted: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, provider restricted: .containers[0].privileged: Invalid value: true: Privileged containers are not allowed, provider restricted: .containers[0].hostNetwork: Invalid value: true: Host network is not allowed to be used, provider restricted: .containers[0].containers[0].hostPort: Invalid value: 3300: Host ports are not allowed to be used, provider restricted: .containers[0].containers[0].hostPort: Invalid value: 6789: Host ports are not allowed to be used, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider rook-ceph: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, provider rook-ceph: .containers[0].hostNetwork: Invalid value: true: Host network is not allowed to be used, provider rook-ceph: .containers[0].containers[0].hostPort: Invalid value: 3300: Host ports are not allowed to be used, provider rook-ceph: .containers[0].containers[0].hostPort: Invalid value: 6789: Host ports are not allowed to be used, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "rook-ceph-csi": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]`

As the deployments are controlled by the rook-ceph-operator I cannot update the deployments manually...

**Comment by bdlink** [comment]
John,

I installed okd 4.13.0-0.okd-2023-07-23-051208 on vSphere UPI and cloned rook 1.12.1 , labeled three worker nodes with role=storage-node and rook/ceph installed with no issues (I had separate raw 1 TB hard disks on each worker node which the rook/ceph formatted).

I then installed rook-ceph-block and rook-cephfs storage classes and created a PVC for the registry (seems ReadWriteMany won’t autoprovision with block storage but itiwill with cephfs).

Everything works fine.

Bruce

**Reply by fortinj66** [reply]
Thanks Bruce...  I think I'm going to rebuild the cluster from scratch try it again...

I'm assuming that you are using the operator-openshift.yaml operator script?  DId you have to add/run any other RBAC scripts?

**Reply by fortinj66** [reply]
Created a new vSphere IPI cluster from 4.13.0-0.okd-2023-09-03-082426.tar.gz and Ceph 1.12.3.  Had the exact same issue...

Vanilla manifests for common.yaml, crds.yaml and operator-openshift.yaml   

```
Error creating: pods "rook-ceph-mon-a-canary-567b88886d-" is forbidden: 
unable to validate against any security context constraint: 
[
provider "anyuid": Forbidden: not usable by user or serviceaccount, 

provider restricted-v2: 
   .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, 
    spec.volumes[2]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
    spec.volumes[3]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
    spec.volumes[4]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
    spec.volumes[5]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
provider restricted-v2: 
    .containers[0].privileged: Invalid value: true: Privileged containers are not allowed, 
provider restricted-v2: 
    .containers[0].hostNetwork: Invalid value: true: Host network is not allowed to be used, 
provider restricted-v2: 
    .containers[0].containers[0].hostPort: Invalid value: 3300: Host ports are not allowed to be used, 
provider restricted-v2: 
    .containers[0].containers[0].hostPort: Invalid value: 6789: Host ports are not allowed to be used, 

provider "restricted": Forbidden: not usable by user or serviceaccount, 
provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, 
provider "nonroot": Forbidden: not usable by user or serviceaccount, 
provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, 
provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, 
provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, 
provider "hostnetwork": Forbidden: not usable by user or serviceaccount, 
provider "hostaccess": Forbidden: not usable by user or serviceaccount, 

provider rook-ceph: 
    .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, 
provider rook-ceph: 
    .containers[0].hostNetwork: Invalid value: true: Host network is not allowed to be used, 
provider rook-ceph: 
    .containers[0].containers[0].hostPort: Invalid value: 3300: Host ports are not allowed to be used, 
provider rook-ceph: 
    .containers[0].containers[0].hostPort: Invalid value: 6789: Host ports are not allowed to be used, 

provider "node-exporter": Forbidden: not usable by user or serviceaccount, 
provider "rook-ceph-csi": Forbidden: not usable by user or serviceaccount, 
provider "privileged": Forbidden: not usable by user or serviceaccount
]
```

**Reply by fortinj66** [reply, marked as answer]
I updated the rook-ceph SCC and made hostNetwork=true and hostPort=true. 

Cluster build proceeded properly with that.
